packages/@lwc/features/src/runtime.ts

```diff
@@ -4,61 +4,17 @@
  * SPDX-License-Identifier: MIT
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
-import { isUndefined } from './shared/language';
+import { create } from './shared/language';
 import { FeatureFlag } from './flags';
 
-function getGlobalThis() {
-    if (typeof globalThis !== 'undefined') {
-        return globalThis;
-    }
-    // globalThis is not available in this env, so let's use some magic
-    // eslint-disable-next-line no-extend-native
-    Object.defineProperty(Object.prototype, '__magic__', {
-        get: function() {
-            return this;
-        },
-        configurable: true,
-    });
-    // @ts-ignore
-    // eslint-disable-next-line no-undef
-    const g = __magic__;
-    // @ts-ignore
-    delete Object.prototype.__magic__;
-    return g;
+interface FeatureFlagConfig {
+    [flagName: string]: FeatureFlag;
 }
 
-const _globalThis = getGlobalThis();
+const runtimeFlags: FeatureFlagConfig = create(null);
 
-// Initialize the configuration object if it isn't initialized already.
-const LWC_config = _globalThis.LWC_config || {};
-
-let runtimeFlags: { [name: string]: FeatureFlag };
-if (process.env.NODE_ENV === 'production') {
-    runtimeFlags = Object.assign({}, LWC_config.features);
-} else {
-    // For testing purposes, we want to share the same runtime flags across
-    // bundles (e.g., engine, synthetic-shadow, etc).
-    runtimeFlags = _globalThis.__lwcRuntimeFlags__;
-    if (isUndefined(runtimeFlags)) {
-        runtimeFlags = Object.assign({}, LWC_config.features);
-        _globalThis.__lwcRuntimeFlags__ = runtimeFlags;
-    }
-}
-
-function enableFeature(name: string) {
-    if (process.env.NODE_ENV === 'production') {
-        // this function should never leak to prod
-        throw new ReferenceError();
-    }
-    runtimeFlags[name] = true;
-}
-
-function disableFeature(name: string) {
-    if (process.env.NODE_ENV === 'production') {
-        // this function should never leak to prod
-        throw new ReferenceError();
-    }
-    runtimeFlags[name] = false;
+function setFeatureFlag(name: string, value: FeatureFlag) {
+    runtimeFlags[name] = value;
```

Comment:
Probably here we should validate that the name is a qualifying flag, and error out if it is not... at least in dev-mode to avoid typos and other things.

Reply:
Flags are currently defined as variables so we would need to maintain a separate array of variable names, or do some refactoring so that flags are defines as entries in an object. The latter option would require a little work it sounds like it might be worth it.

Reply:
If we want to do the above and this PR looks good otherwise, we can merge it for now to unblock the reactive setter changes and I can work on the flag changes a little later.

Reply:
ok, good!